src/ImageStats/Histogram.cc

```diff
@@ -13,33 +13,52 @@
 
 using namespace carta;
 
-Histogram::Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data)
-    : _num_bins(num_bins), _bin_width((max_value - min_value) / num_bins), _min_val(min_value), _hist(num_bins, 0), _data(data) {}
+Histogram::Histogram(int num_bins, float min_value, float max_value, const std::vector<float>& data) :_data(data) {
+    _histogram.num_bins = num_bins;
+    _histogram.bin_width =  (max_value - min_value) / num_bins;
+    _min_val = min_value;
+    _max_val = max_value;
+    _histogram.histogram_bins.resize(num_bins);
+    _histogram.bin_center = _min_val + (_histogram.bin_width * 0.5);
+}
 
-Histogram::Histogram(Histogram& h, tbb::split)
-    : _num_bins(h._num_bins), _bin_width(h._bin_width), _min_val(h._min_val), _hist(h._hist.size(), 0), _data(h._data) {}
+Histogram::Histogram(Histogram& h, tbb::split) : _data(h.GetData()) {
+    _histogram.num_bins = h.GetNbins();
+    _histogram.bin_width = h.GetBinWidth();
+    _min_val = h.GetMinVal();
+    _max_val = h.GetMaxVal();
+    _histogram.histogram_bins = h.GetHistogramBins();
+}
 
 void Histogram::operator()(const tbb::blocked_range<size_t>& r) {
-    std::vector<int> tmp(_hist);
+    std::vector<int> tmp(_histogram.histogram_bins);
     for (auto i = r.begin(); i != r.end(); ++i) {
         auto v = _data[i];
         if (std::isnan(v) || std::isinf(v)) {
             continue;
         }
-
-        size_t bin_number = std::max<size_t>(std::min<size_t>((size_t)((v - _min_val) / _bin_width), (size_t)_hist.size() - 1), 0);
-
+        size_t bin_number = std::max<size_t>(std::min<size_t>((size_t)((v - _min_val) / _histogram.bin_width), (size_t)_histogram.histogram_bins.size() - 1), 0);
         ++tmp[bin_number];
     }
-    _hist = tmp;
+    _histogram.histogram_bins = tmp;
 }
 
-void Histogram::join(Histogram& h) { // NOLINT
-    auto num_bins = h._hist.size();
+bool Histogram::join(Histogram& h) { // NOLINT
+    /** TODO:
+     * add a consitency check before moving on...
+     * check if the other object exists
+     */
+    // ConsistencyCheck(this, h);
+    // if(!h)
+    const int num_bins = h.GetHistogramBins().size();
+    if (this->GetHistogramBins().size() != num_bins) {
+        return false;
+    }
 #pragma omp parallel for
     for (int i = 0; i < num_bins; i++) {
-        h._hist[i] += _hist[i];
+        _histogram.histogram_bins[i] += h.GetHistogramBins()[i];
```

Comment:
I think it would be better to move `h.GetHistogramBins()` out of the parallel for (just using `auto& other_bins = h.GetHistogramBins()`

Reply:
Sure, good call.